src/gui/torrentcontentmodel.cpp

```diff
@@ -595,49 +592,11 @@ void TorrentContentModel::notifySubtreeUpdated(const QModelIndex &index, const Q
 {
     // For best performance, `columns` entries should be arranged from left to right
 
-    Q_ASSERT(index.isValid());
+    Q_UNUSED(index);
 
-    // emit itself
+    // Hack!
+    // Emitting dataChanged() for a large tree takes considerable time and therefore just trick the View to reload data
+    // For original implementation, refer to: https://github.com/qbittorrent/qBittorrent/blob/7afe34fd8986b31b28e19808cd580bcbff8350a7/src/gui/torrentcontentmodel.cpp#L594-L643
     for (const ColumnInterval &column : columns)
-        emit dataChanged(index.siblingAtColumn(column.first()), index.siblingAtColumn(column.last()));
-
-    // propagate up the model
-    QModelIndex parentIndex = parent(index);
-    while (parentIndex.isValid())
-    {
-        for (const ColumnInterval &column : columns)
-            emit dataChanged(parentIndex.siblingAtColumn(column.first()), parentIndex.siblingAtColumn(column.last()));
-        parentIndex = parent(parentIndex);
-    }
-
-    // propagate down the model
-    QVector<QModelIndex> parentIndexes;
-
-    if (hasChildren(index))
-        parentIndexes.push_back(index);
-
-    while (!parentIndexes.isEmpty())
-    {
-        const QModelIndex parent = parentIndexes.takeLast();
-
-        const int childCount = rowCount(parent);
-        const QModelIndex child = this->index(0, 0, parent);
-
-        // emit this generation
-        for (const ColumnInterval &column : columns)
-        {
-            const QModelIndex childTopLeft = child.siblingAtColumn(column.first());
-            const QModelIndex childBottomRight = child.sibling((childCount - 1), column.last());
-            emit dataChanged(childTopLeft, childBottomRight);
-        }
-
-        // check generations further down
-        parentIndexes.reserve(childCount);
-        for (int i = 0; i < childCount; ++i)
-        {
-            const QModelIndex sibling = child.siblingAtRow(i);
-            if (hasChildren(sibling))
-                parentIndexes.push_back(sibling);
-        }
-    }
+        emit headerDataChanged(Qt::Horizontal, column.first(), column.last());
```

Comment:
This might be controversial but I don't have better ideas. All other alternatives either have some pitfalls or they won't work reliably.
Also I'm ok if this hack is not accepted.

Reply:
>they won't work reliably.

Is this workaround can be considered as working any reliable?

Reply:
>Is this workaround can be considered as working any reliable?

At least it work for me in Qt5 & 6 and linux & windows.

Reply:
> > Is this workaround can be considered as working any reliable?
> 
> At least it work for me in Qt5 & 6 and linux & windows.

This is a pure workaround, because no one promises that when processing this signal, the View will update something other than the column headers themselves.
I have no fundamental objections to using it if it has a tangible performance gain. But this will at least require us to be extra vigilant with every Qt update.

Reply:
>This is a pure workaround, because no one promises that when processing this signal, the View will update something other than the column headers themselves.

Exactly. jagannatharjun will need to confirm the performance gain in order to justify the hack.

>I have no fundamental objections to using it if it has a tangible performance gain. But this will at least require us to be extra vigilant with every Qt update.

That is why I put the code comment so it can be less painful when reverting.